drt: add monitoring scripts and dashboards

herkolategan opened the pull request:
Epic: None
Release Note: None

herkolategan committed:
drt: add prometheus config generator script

Epic: None
Release Note: None

herkolategan committed:
drt: add resilience testing grafana dashboard json

Epic: None
Release Note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/120695)
<!-- Reviewable:end -->

herkolategan committed:
drt: update CODEOWNERS pattern

Epic: None
Release Note: None

herkolategan committed:
drt: gen prom config secure flag

Setting `SECURE_ONLY` to `true` will prevent scraping `http` on nodes in the
project. This avoids noisy TLS errors in the cockroach logs.

Epic: None
Release Note: None

itsbilal reviewed (approved):
LGTM, thanks for doing this!

herkolategan commented on pkg/cmd/drt/monitoring/prometheus/generate_config.sh:
```diff
@@ -0,0 +1,279 @@
+
+# Cockroach metrics are collected from port 26258, we only replace the region here
+prom_cockroach=$(cat <<'EOF'
+- job_name: 'crdb-{project}-{region}'
```
@itsbilal True, it's why I added the `ONLY_SECURE` flag, which is true by default and will ignore this whole section when generating the config. But maybe I should remove the flag and this section if we're sure we will always only have secure clusters for DRT - was thinking in terms of general prom config and not DRT specific.

herkolategan commented:
TFTR!

bors r=itsbilal
